### DISCUSSION TITLE
Raspberry Pi 5 GPIO question

### DISCUSSION THREAD

**Discussion by cookin83** [paukstelis/octoprint_deploy#175] [Q&A]
On my Raspberry Pi 2 I used the Autoprint plugin to power on/off my printer. I've deployed octoprint on a Raspberry Pi 5 but can't seem to get Autoprint to control the relay I have connected to GPIO 18. I have double and triple checked the wiring and tried different GPIO pins but can't seem to get it to work. Is there something I need to do in the octoprint_deploy settings to allow octoprint to use the GPIO pins?

**Comment by paukstelis** [comment]
No, there is nothing special in octoprint_deploy. Typically, I think you would need to enable GPIO on the Pi itself. likely through raspi-config.

**Comment by cookin83** [comment]
After playing around with a couple python scripts it looks like the old autoprint plugin uses import RPi.GPIO as GPIO at the beginning of the script to access the GPIO pins but on a Raspberry Pi 5 it needs to use import gpiod. I found the python script in the plugin folder and tried changing the import line but when I do the plugin disappears from octoprint. I tried cloning, editing, and reinstalling the plugin but still can't get it to show up.
